File: Python Sample/greatest_common_divisor_of_strings.py
import math

class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:

        if (str1+str2) != (str2+str1):
            return ""

        l1 = len(str1)
        l2 = len(str2)

        if l1 == l2 and str1 == str2:
            return str1
        else:
            gcd = math.gcd(l1, l2)
            print(f"gcd:{gcd}")
            return str2[0:gcd]

File: Go smple/internal/Greatest Common Divisor of Strings.go
package internal

func gcdOfStrings(str1 string, str2 string) string {

    l1 := len(str1)
    l2 := len(str2)

    if (str1 + str2) != (str2 + str1){

        return ""
    } else {
       
        if l1 == l2 && str1 == str2 {
            return str1
        } else {
            
            gcd := gcd(l1, l2)
            fmt.Println(gcd)
            return str2[:gcd]
        }
        return ""
    }
    
}

func gcd(a,b int) int {
	if b == 0 {
		return a
	}
	return gcd(b, a % b)
}